File: Lambda/src/main/java/me/denismurphy/tranquil/handlers/PauseIntentHandler.java
package me.denismurphy.tranquil.handlers;

import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.dispatcher.request.handler.RequestHandler;
import com.amazon.ask.model.Response;
import com.amazon.ask.request.Predicates;

import java.util.Optional;

public class PauseIntentHandler implements RequestHandler {

    @Override
    public boolean canHandle(HandlerInput input) {
        return input.matches(Predicates.intentName("AMAZON.PauseIntent"));
    }

    @Override
    public Optional<Response> handle(HandlerInput input) {
        input.getAttributesManager().getPersistentAttributes().put("offset", input.getRequestEnvelope().getContext().getAudioPlayer().getOffsetInMilliseconds());
        input.getAttributesManager().savePersistentAttributes();

        return input.getResponseBuilder()
                .addAudioPlayerStopDirective()
                .build();
    }
}

File: lambda/src/main/kotlin/me/denismurphy/tranquil/handlers/PauseIntentHandler.kt
package me.denismurphy.tranquil.handlers

import com.amazon.ask.dispatcher.request.handler.HandlerInput
import com.amazon.ask.dispatcher.request.handler.RequestHandler
import com.amazon.ask.model.Response
import com.amazon.ask.request.Predicates
import java.util.*

class PauseIntentHandler : RequestHandler {

    override fun canHandle(input: HandlerInput): Boolean {
        return input.matches(Predicates.intentName("AMAZON.PauseIntent"))
    }

    override fun handle(input: HandlerInput): Optional<Response> {
        input.attributesManager.persistentAttributes["offset"] = input.requestEnvelope.context.audioPlayer.offsetInMilliseconds
        input.attributesManager.savePersistentAttributes()

        return input.responseBuilder
                .addAudioPlayerStopDirective()
                .build()
    }
}